module/util/context.go

```diff
@@ -0,0 +1,106 @@
+package util
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+)
+
+var ErrChannelClosed = errors.New("channel closed")
+var ErrSignalReceived = errors.New("signal received")
+
+// WithDone wraps a signal channel with a context, and cancels the context when the channel is closed.
+// When the context is Done, the ctx.Err() will either be ErrChannelClosed if the channel closed first,
+// or the error from the underlying context (Canceled, DeadlineExceeded, etc).
+func WithDone(parent context.Context, done <-chan struct{}) (context.Context, context.CancelFunc) {
```

Comment:
functions added in `module/util/context.go` and `module/util/util.go` are missing unit test coverage.

Reply:
added in 2e555f6